lib/std/Progress.zig

```diff
@@ -380,16 +447,36 @@ pub fn unlock_stderr(p: *Progress) void {
     p.update_mutex.unlock();
 }
 
+/// Wrapping function around `bufWrite` that cares about the terminal width
+fn bufWriteLineTruncate(self: *Progress, end: *usize, comptime format: []const u8, args: anytype) void {
+    comptime assert(std.mem.count(u8, format, "\n") == 0);
+    const ellipse = "...";
+    const start_index = end.*;
+    if (self.max_columns <= 0 or self.columns_written < self.max_columns) self.bufWrite(end, format, args);
+    const actual_columns_written = end.* - start_index;
+    const truncated_end_index = start_index + @min(self.max_columns - self.columns_written, actual_columns_written);
+    if (self.max_columns > 0) end.* = truncated_end_index;
+    self.columns_written += truncated_end_index - start_index;
+    if (self.max_columns > 0 and actual_columns_written + self.columns_written > self.max_columns) {
+        @memcpy(self.output_buffer[end.* - ellipse.len .. end.*], ellipse);
```

Comment:
I am pretty sure that if `end` is smaller than the ellipse length, this code will fail. A check could prevent this.